kong/conf_loader.lua

```diff
@@ -357,6 +389,99 @@ local function check_and_infer(conf, opts)
     end
   end
 
+  for _, name in ipairs({
+    "cassandra_consistency_proxy",
+    "cassandra_consistency_admin"
+  }) do
+    local read_consistency
+    local write_consistency
+    local option = conf[name]
+    local number_of_values = option and #option or 0
+    if number_of_values > 2 then
+      if is_cassandra then
+        errors[#errors + 1] = fmt("%s has an invalid value: '%s' (%s)", name,
+                                  concat(option, ", "),
+                                  "specify at most two values, e.g. " ..
+                                  "'read:ONE, write:ONE', or just one " ..
+                                  "for both e.g. 'QUORUM'")
+      end
+
+    elseif number_of_values == 2 then
+      read_consistency = CASSANDRA_READ_CONSISTENCY[option[1]] or
+                         CASSANDRA_READ_CONSISTENCY[option[2]]
+
+      write_consistency = CASSANDRA_WRITE_CONSISTENCY[option[1]] or
+                          CASSANDRA_WRITE_CONSISTENCY[option[2]]
+
+      if not read_consistency or not write_consistency and is_cassandra then
+        errors[#errors + 1] = fmt("%s has an invalid value: '%s' (read:%s, " ..
+                                  "write:%s)", name, concat(option, ", "),
+                                  concat(CASSANDRA_CONSISTENCY, ", read:"),
+                                  concat(CASSANDRA_CONSISTENCY, ", write:"))
+      end
+
+    elseif number_of_values == 1 then
+      local found
+      for _, name in ipairs(CASSANDRA_CONSISTENCY) do
+        if name == option[1] then
+          found = true
+        end
+      end
+
+      if not found then
+        if is_cassandra then
+          errors[#errors + 1] = fmt("%s has an invalid value: '%s' (%s)", name,
+                                    option[1],
+                                    concat(CASSANDRA_CONSISTENCY, ", "))
+        end
+
+      else
+        read_consistency  = option[1]
+        write_consistency = option[1]
+      end
+    end
+
+    if not read_consistency then
+      read_consistency  = conf.cassandra_consistency
+    end
+
+    if not write_consistency then
+      write_consistency = conf.cassandra_consistency
+    end
```

Comment:
Why do we have those two fallbacks? 

- If the user specified `cassandra_consistency` and none of those above two properties, the deprecated alias should assign `_admin` and `_proxy` (and log the warning).
- If the user specified did not specify `cassandra_consistency`, then he or she removed it from a config file or env variable, and we should fully rely on the new `_admin` and `_proxy` settings (since they have defaults and we removed `cassandra_consistency` from the defaults list as well).

What I am trying to say is that if specified, the _deprecated_ property should **still have precedence** over the new one(s), and the warning logs should be printed. As soon as the user removes the deprecated property, then the new properties reign supreme, and their validation rules apply, with no fallback whatsoever to deprecated properties.

This feels wrong here. `_proxy` and `_admin` should not be allowed to be empty in this code path without `errors` being populated. This fallback should only be in the `alias` function, and invoked _before_ `check_and_infer()`.